WASM build is broken :(

On `$ cargo make wasm` I get:

```
error: target is not supported, for more information see: https://docs.rs/getrandom/#unsupported-targets
   --> /Users/sffc/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.2.2/src/lib.rs:213:9
    |
213 | /         compile_error!("target is not supported, for more information see: \
214 | |                         https://docs.rs/getrandom/#unsupported-targets");
    | |_________________________________________________________________________^

error[E0433]: failed to resolve: use of undeclared crate or module `imp`
   --> /Users/sffc/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.2.2/src/lib.rs:235:5
    |
235 |     imp::getrandom_inner(dest)
    |     ^^^ use of undeclared crate or module `imp`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0433`.
error: could not compile `getrandom`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
[cargo-make] ERROR - Error while executing command, exit code: 101
[cargo-make] WARN - Build Failed.
```

I guess the version of `getrandom` got bumped at some point and added this error.  I will look into how to fix it.

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -37,7 +37,7 @@ bincode-gen-testdata = "make bincode-gen-testdata"
 ### WASM TASKS ###
 
 # Re-build standard library with panic=abort.
-wasm-build = "build -Z build-std=std,panic_abort --target wasm32-unknown-unknown --release"
+wasm-build = "build -Z build-std=std,panic_abort -Z build-std-features=panic_immediate_abort --target wasm32-unknown-unknown --release"
 
 [target.wasm32-unknown-unknown]
 rustflags = [

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -280,6 +280,7 @@ jobs:
               icu_uniset/unicode_bmp_blocks_selector
               fixed_decimal/permyriad
               writeable/writeable_message
+              litemap/language_names_lite_map
 
     runs-on: ${{ matrix.os }}
 

--- a/Makefile.toml
+++ b/Makefile.toml
@@ -98,7 +98,7 @@ dependencies = [
 description = "Build all examples as WASM into the target directory"
 category = "ICU4X WASM"
 install_crate = { rustup_component_name = "rust-src" }
-toolchain = "nightly"
+toolchain = "nightly-2021-03-15"
 command = "cargo"
 args = ["wasm-build", "--examples"]
 
@@ -118,26 +118,65 @@ dependencies = ["wasm-build", "wasm-dir"]
 [tasks.wasm-wat]
 description = "Create WebAssembly Text files from the WASM files"
 category = "ICU4X WASM"
-command = "find"
-args = ["wasmpkg/", "-name", "*.wasm", "-exec", "wasm2wat", "{}", "-o", "{}.wat", ";"]
 dependencies = ["wasm-wasm"]
-install_script = ["which wasm2wat || npm install -g wabt"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+wasm2wat = which wasm2wat
+assert ${wasm2wat} "Could not find 'wasm2wat' in path.\n*** Please run 'npm install -g wabt' ***"
+
+handle = glob_array wasmpkg/*.wasm
+for path in ${handle}
+    basename = substring ${path} -5
+    out_path = concat ${basename} ".wat"
+    exec --fail-on-error ${wasm2wat} ${path} -o ${out_path}
+end
+'''
 
 [tasks.wasm-opt]
 description = "Create optimized WASM files from the WASM files"
 category = "ICU4X WASM"
-command = "find"
-args = ["wasmpkg/", "-name", "*.wasm", "-exec", "wasm-opt", "{}", "-o", "{}.opt", ";"]
 dependencies = ["wasm-wasm"]
-install_script = ["which wasm-opt || npm install -g wasm-opt"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+wasm-opt = which wasm-opt
+assert ${wasm-opt} "Could not find 'wasm-opt' in path.\n*** Please run 'npm install -g wasm-opt' ***"
+
+handle = glob_array wasmpkg/*.wasm
+for path in ${handle}
+    basename = substring ${path} -5
+    out_path = concat ${basename} "+opt.wasm"
+    output = exec ${wasm-opt} ${path} -o ${out_path}
+    stdout = trim ${output.stdout}
+    stderr = trim ${output.stderr}
+    if ${stdout} or ${stderr} or ${output.code}
+        echo ${stdout}\n${stderr}\nexit code: ${output.code}
+        assert_fail "wasm-opt printed warnings (shown above)"
+    end
+end
+'''
 
 [tasks.wasm-twiggy-dominators]
 description = "Create Twiggy Dominator files from the WASM files"
 category = "ICU4X WASM"
-command = "find"
-args = ["wasmpkg/", "-name", "*.wasm", "-exec", "twiggy", "dominators", "{}", "-o", "{}.txt", ";"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+twiggy = which twiggy
+assert ${twiggy} "Could not find 'twiggy' in path.\n*** Please run 'cargo install twiggy' ***"
+
+handle = glob_array wasmpkg/*.wasm
+for path in ${handle}
+    basename = substring ${path} -5
+    out_path = concat ${basename} "+twiggy.txt"
+    exec --fail-on-error ${twiggy} dominators ${path} -o ${out_path}
+end
+'''
 dependencies = ["wasm-wasm"]
-install_crate = "twiggy"
 
 [tasks.wasm]
 description = "All-in-one command to build examples and supplements to wasmpkg"

--- a/Makefile.toml
+++ b/Makefile.toml
@@ -155,6 +155,46 @@ for src_path in ${handle}
 end
 '''
 
+[tasks.wasm-dcmp]
+description = "Create WebAssembly decompiled code files from the WASM files"
+category = "ICU4X WASM"
+dependencies = ["wasm-wasm"]
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+wasm-decompile = which wasm-decompile
+assert ${wasm-decompile} "Could not find 'wasm-decompile' in path.\n*** Please run 'npm install -g wabt' ***"
+
+mkdir wasmpkg/wasm-decompile
+
+handle = glob_array wasmpkg/*.wasm
+for src_path in ${handle}
+    path_no_extension = substring ${src_path} -5
+    basename = substring ${path_no_extension} 8
+    out_path = concat wasmpkg/wasm-decompile/ ${basename} ".dcmp"
+
+    out_exists = is_path_exists ${out_path}
+    up_to_date = set false
+    if ${out_exists}
+        src_time = get_last_modified_time ${src_path}
+        out_time = get_last_modified_time ${out_path}
+        up_to_date = less_than ${src_time} ${out_time}
+    end
+
+    if not ${up_to_date}
+        echo Writing ${out_path}
+        output = exec ${wasm-decompile} ${src_path} -o ${out_path}
+        stdout = trim ${output.stdout}
+        stderr = trim ${output.stderr}
+        if ${stdout} or ${stderr} or ${output.code}
+            echo ${stdout}\n${stderr}\nexit code: ${output.code}
+            assert_fail "wasm-decompile printed warnings (shown above)"
+        end
+    end
+end
+'''
+
 [tasks.wasm-opt]
 description = "Create optimized WASM files from the WASM files"
 category = "ICU4X WASM"
@@ -235,6 +275,7 @@ category = "ICU4X WASM"
 dependencies = [
     "wasm-wasm",
     "wasm-wat",
+    "wasm-dcmp",
     "wasm-opt",
     "wasm-twiggy-dominators",
 ]

--- a/Makefile.toml
+++ b/Makefile.toml
@@ -98,7 +98,7 @@ dependencies = [
 description = "Build all examples as WASM into the target directory"
 category = "ICU4X WASM"
 install_crate = { rustup_component_name = "rust-src" }
-toolchain = "nightly-2021-03-15"
+toolchain = "nightly-2021-02-28"
 command = "cargo"
 args = ["wasm-build", "--examples"]
 
@@ -126,11 +126,32 @@ exit_on_error true
 wasm2wat = which wasm2wat
 assert ${wasm2wat} "Could not find 'wasm2wat' in path.\n*** Please run 'npm install -g wabt' ***"
 
+mkdir wasmpkg/wat
+
 handle = glob_array wasmpkg/*.wasm
-for path in ${handle}
-    basename = substring ${path} -5
-    out_path = concat ${basename} ".wat"
-    exec --fail-on-error ${wasm2wat} ${path} -o ${out_path}
+for src_path in ${handle}
+    path_no_extension = substring ${src_path} -5
+    basename = substring ${path_no_extension} 8
+    out_path = concat wasmpkg/wat/ ${basename} ".wat"
+
+    out_exists = is_path_exists ${out_path}
+    up_to_date = set false
+    if ${out_exists}
+        src_time = get_last_modified_time ${src_path}
+        out_time = get_last_modified_time ${out_path}
+        up_to_date = less_than ${src_time} ${out_time}
+    end
+
+    if not ${up_to_date}
+        echo Writing ${out_path}
+        output = exec ${wasm2wat} ${src_path} -o ${out_path}
+        stdout = trim ${output.stdout}
+        stderr = trim ${output.stderr}
+        if ${stdout} or ${stderr} or ${output.code}
+            echo ${stdout}\n${stderr}\nexit code: ${output.code}
+            assert_fail "wasm2wat printed warnings (shown above)"
+        end
+    end
 end
 '''
 
@@ -145,16 +166,31 @@ exit_on_error true
 wasm-opt = which wasm-opt
 assert ${wasm-opt} "Could not find 'wasm-opt' in path.\n*** Please run 'npm install -g wasm-opt' ***"
 
+mkdir wasmpkg/wasm-opt
+
 handle = glob_array wasmpkg/*.wasm
-for path in ${handle}
-    basename = substring ${path} -5
-    out_path = concat ${basename} "+opt.wasm"
-    output = exec ${wasm-opt} ${path} -o ${out_path}
-    stdout = trim ${output.stdout}
-    stderr = trim ${output.stderr}
-    if ${stdout} or ${stderr} or ${output.code}
-        echo ${stdout}\n${stderr}\nexit code: ${output.code}
-        assert_fail "wasm-opt printed warnings (shown above)"
+for src_path in ${handle}
+    path_no_extension = substring ${src_path} -5
+    basename = substring ${path_no_extension} 8
+    out_path = concat wasmpkg/wasm-opt/ ${basename} "+opt.wasm"
+
+    out_exists = is_path_exists ${out_path}
+    up_to_date = set false
+    if ${out_exists}
+        src_time = get_last_modified_time ${src_path}
+        out_time = get_last_modified_time ${out_path}
+        up_to_date = less_than ${src_time} ${out_time}
+    end
+
+    if not ${up_to_date}
+        echo Writing ${out_path}
+        output = exec ${wasm-opt} ${src_path} -o ${out_path}
+        stdout = trim ${output.stdout}
+        stderr = trim ${output.stderr}
+        if ${stdout} or ${stderr} or ${output.code}
+            echo ${stdout}\n${stderr}\nexit code: ${output.code}
+            assert_fail "wasm-opt printed warnings (shown above)"
+        end
     end
 end
 '''
@@ -169,11 +205,26 @@ exit_on_error true
 twiggy = which twiggy
 assert ${twiggy} "Could not find 'twiggy' in path.\n*** Please run 'cargo install twiggy' ***"
 
+mkdir wasmpkg/twiggy
+
 handle = glob_array wasmpkg/*.wasm
-for path in ${handle}
-    basename = substring ${path} -5
-    out_path = concat ${basename} "+twiggy.txt"
-    exec --fail-on-error ${twiggy} dominators ${path} -o ${out_path}
+for src_path in ${handle}
+    path_no_extension = substring ${src_path} -5
+    basename = substring ${path_no_extension} 8
+    out_path = concat wasmpkg/twiggy/ ${basename} "+twiggy.txt"
+
+    out_exists = is_path_exists ${out_path}
+    up_to_date = set false
+    if ${out_exists}
+        src_time = get_last_modified_time ${src_path}
+        out_time = get_last_modified_time ${out_path}
+        up_to_date = less_than ${src_time} ${out_time}
+    end
+
+    if not ${up_to_date}
+        echo Writing ${out_path}
+        exec --fail-on-error ${twiggy} dominators ${src_path} -o ${out_path}
+    end
 end
 '''
 dependencies = ["wasm-wasm"]

--- a/components/decimal/Cargo.toml
+++ b/components/decimal/Cargo.toml
@@ -34,9 +34,10 @@ criterion = "0.3"
 icu_benchmark_macros = { version = "0.1", path = "../../tools/benchmark/macros" }
 icu_locid_macros = { version = "0.1", path = "../locid/macros" }
 icu_testdata = { version = "0.1", path = "../../resources/testdata" }
-rand = "0.7"
-rand_pcg = "0.2"
-rand_distr = "0.3"
+rand = "0.8"
+rand_pcg = "0.3"
+rand_distr = "0.4"
+getrandom = { version = "0.2", features = ["js"] }
 
 [features]
 default = ["provider_serde"]

--- a/components/decimal/Cargo.toml
+++ b/components/decimal/Cargo.toml
@@ -34,9 +34,9 @@ criterion = "0.3"
 icu_benchmark_macros = { version = "0.1", path = "../../tools/benchmark/macros" }
 icu_locid_macros = { version = "0.1", path = "../locid/macros" }
 icu_testdata = { version = "0.1", path = "../../resources/testdata" }
-rand = "0.8"
-rand_pcg = "0.3"
-rand_distr = "0.4"
+rand = "0.7"
+rand_pcg = "0.2"
+rand_distr = "0.3"
 
 [features]
 default = ["provider_serde"]

--- a/utils/fixed_decimal/Cargo.toml
+++ b/utils/fixed_decimal/Cargo.toml
@@ -28,9 +28,9 @@ writeable = { version = "0.2", path = "../../utils/writeable" }
 [dev-dependencies]
 criterion = "0.3.3"
 icu_benchmark_macros = { version = "0.1", path = "../../tools/benchmark/macros" }
-rand = "0.8"
-rand_pcg = "0.3"
-rand_distr = "0.4"
+rand = "0.7"
+rand_pcg = "0.2"
+rand_distr = "0.3"
 
 [lib]
 bench = false  # This option is required for Benchmark CI

--- a/utils/fixed_decimal/Cargo.toml
+++ b/utils/fixed_decimal/Cargo.toml
@@ -28,9 +28,10 @@ writeable = { version = "0.2", path = "../../utils/writeable" }
 [dev-dependencies]
 criterion = "0.3.3"
 icu_benchmark_macros = { version = "0.1", path = "../../tools/benchmark/macros" }
-rand = "0.7"
-rand_pcg = "0.2"
-rand_distr = "0.3"
+rand = "0.8"
+rand_pcg = "0.3"
+rand_distr = "0.4"
+getrandom = { version = "0.2", features = ["js"] }
 
 [lib]
 bench = false  # This option is required for Benchmark CI

--- a/utils/litemap/Cargo.toml
+++ b/utils/litemap/Cargo.toml
@@ -21,6 +21,9 @@ serde = {version = "1", optional = true}
 [dev-dependencies]
 serde = "1"
 serde_json = "1"
+icu_locid = { version = "0.1", path = "../../components/locid" }
+icu_locid_macros = { version = "0.1", path = "../../components/locid/macros" }
+icu_benchmark_macros = { version = "0.1", path = "../../tools/benchmark/macros" }
 
 [[test]]
 name = "serde"

--- a/utils/litemap/examples/language_names_hash_map.rs
+++ b/utils/litemap/examples/language_names_hash_map.rs
@@ -9,8 +9,23 @@
 
 icu_benchmark_macros::static_setup!();
 
-use std::collections::HashMap;
+use icu_locid::subtags::Language;
 use icu_locid_macros::language;
+use std::collections::HashMap;
+
+const DATA: [(Language, &'static str); 11] = [
+    (language!("ar"), "Arabic"),
+    (language!("bn"), "Bangla"),
+    (language!("ccp"), "Chakma"),
+    (language!("en"), "English"),
+    (language!("es"), "Spanish"),
+    (language!("fr"), "French"),
+    (language!("ja"), "Japanese"),
+    (language!("ru"), "Russian"),
+    (language!("sr"), "Serbian"),
+    (language!("th"), "Thai"),
+    (language!("tr"), "Turkish"),
+];
 
 #[no_mangle]
 fn main(_argc: isize, _argv: *const *const u8) -> isize {
@@ -18,20 +33,12 @@ fn main(_argc: isize, _argv: *const *const u8) -> isize {
 
     let mut map = HashMap::new();
     // https://github.com/rust-lang/rust/issues/62633 was declined :(
-    map.insert(language!("ar"), "Arabic").ok_or(()).unwrap_err();
-    map.insert(language!("bn"), "Bangla").ok_or(()).unwrap_err();
-    map.insert(language!("ccp"), "Chakma").ok_or(()).unwrap_err();
-    map.insert(language!("en"), "English").ok_or(()).unwrap_err();
-    map.insert(language!("es"), "Spanish").ok_or(()).unwrap_err();
-    map.insert(language!("fr"), "French").ok_or(()).unwrap_err();
-    map.insert(language!("ja"), "Japanese").ok_or(()).unwrap_err();
-    map.insert(language!("ru"), "Russian").ok_or(()).unwrap_err();
-    map.insert(language!("sr"), "Serbian").ok_or(()).unwrap_err();
-    map.insert(language!("th"), "Thai").ok_or(()).unwrap_err();
-    map.insert(language!("tr"), "Turkish").ok_or(()).unwrap_err();
+    for (lang, name) in DATA.iter() {
+        map.insert(lang, name).ok_or(()).unwrap_err();
+    }
 
     debug_assert_eq!(11, map.len());
-    debug_assert_eq!(Some(&"Thai"), map.get(&language!("th")));
+    debug_assert_eq!(Some(&&"Thai"), map.get(&language!("th")));
     debug_assert_eq!(None, map.get(&language!("de")));
 
     0

--- a/utils/litemap/examples/language_names_hash_map.rs
+++ b/utils/litemap/examples/language_names_hash_map.rs
@@ -1,0 +1,38 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// LiteMap is intended as a small and low-memory drop-in replacement for
+// HashMap. This example demonstrates how it compares to HashMap.
+
+#![no_main] // https://github.com/unicode-org/icu4x/issues/395
+
+icu_benchmark_macros::static_setup!();
+
+use std::collections::HashMap;
+use icu_locid_macros::language;
+
+#[no_mangle]
+fn main(_argc: isize, _argv: *const *const u8) -> isize {
+    icu_benchmark_macros::main_setup!();
+
+    let mut map = HashMap::new();
+    // https://github.com/rust-lang/rust/issues/62633 was declined :(
+    map.insert(language!("ar"), "Arabic").ok_or(()).unwrap_err();
+    map.insert(language!("bn"), "Bangla").ok_or(()).unwrap_err();
+    map.insert(language!("ccp"), "Chakma").ok_or(()).unwrap_err();
+    map.insert(language!("en"), "English").ok_or(()).unwrap_err();
+    map.insert(language!("es"), "Spanish").ok_or(()).unwrap_err();
+    map.insert(language!("fr"), "French").ok_or(()).unwrap_err();
+    map.insert(language!("ja"), "Japanese").ok_or(()).unwrap_err();
+    map.insert(language!("ru"), "Russian").ok_or(()).unwrap_err();
+    map.insert(language!("sr"), "Serbian").ok_or(()).unwrap_err();
+    map.insert(language!("th"), "Thai").ok_or(()).unwrap_err();
+    map.insert(language!("tr"), "Turkish").ok_or(()).unwrap_err();
+
+    debug_assert_eq!(11, map.len());
+    debug_assert_eq!(Some(&"Thai"), map.get(&language!("th")));
+    debug_assert_eq!(None, map.get(&language!("de")));
+
+    0
+}

--- a/utils/litemap/examples/language_names_hash_map.rs
+++ b/utils/litemap/examples/language_names_hash_map.rs
@@ -2,8 +2,10 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-// LiteMap is intended as a small and low-memory drop-in replacement for
-// HashMap. This example demonstrates how it compares to HashMap.
+// LiteMap is intended as a small and low-memory drop-in replacement for HashMap.
+//
+// This example does not use LiteMap. The reader may compare this HashMap example to the
+// LiteMap example to showcase analogous operations between HashMap and LiteMap.
 
 #![no_main] // https://github.com/unicode-org/icu4x/issues/395
 

--- a/utils/litemap/examples/language_names_lite_map.rs
+++ b/utils/litemap/examples/language_names_lite_map.rs
@@ -1,0 +1,38 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// LiteMap is intended as a small and low-memory drop-in replacement for
+// HashMap. This example demonstrates its use.
+
+#![no_main] // https://github.com/unicode-org/icu4x/issues/395
+
+icu_benchmark_macros::static_setup!();
+
+use litemap::LiteMap;
+use icu_locid_macros::language;
+
+#[no_mangle]
+fn main(_argc: isize, _argv: *const *const u8) -> isize {
+    icu_benchmark_macros::main_setup!();
+
+    let mut map = LiteMap::new();
+    // https://github.com/rust-lang/rust/issues/62633 was declined :(
+    map.try_append(language!("ar"), "Arabic").ok_or(()).unwrap_err();
+    map.try_append(language!("bn"), "Bangla").ok_or(()).unwrap_err();
+    map.try_append(language!("ccp"), "Chakma").ok_or(()).unwrap_err();
+    map.try_append(language!("en"), "English").ok_or(()).unwrap_err();
+    map.try_append(language!("es"), "Spanish").ok_or(()).unwrap_err();
+    map.try_append(language!("fr"), "French").ok_or(()).unwrap_err();
+    map.try_append(language!("ja"), "Japanese").ok_or(()).unwrap_err();
+    map.try_append(language!("ru"), "Russian").ok_or(()).unwrap_err();
+    map.try_append(language!("sr"), "Serbian").ok_or(()).unwrap_err();
+    map.try_append(language!("th"), "Thai").ok_or(()).unwrap_err();
+    map.try_append(language!("tr"), "Turkish").ok_or(()).unwrap_err();
+
+    debug_assert_eq!(11, map.len());
+    debug_assert_eq!(Some(&"Thai"), map.get(&language!("th")));
+    debug_assert_eq!(None, map.get(&language!("de")));
+
+    0
+}

--- a/utils/litemap/examples/language_names_lite_map.rs
+++ b/utils/litemap/examples/language_names_lite_map.rs
@@ -2,8 +2,10 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-// LiteMap is intended as a small and low-memory drop-in replacement for
-// HashMap. This example demonstrates its use.
+// LiteMap is intended as a small and low-memory drop-in replacement for HashMap.
+//
+// The reader may compare this LiteMap example with the HashMap example to see analogous
+// operations between LiteMap and HashMap.
 
 #![no_main] // https://github.com/unicode-org/icu4x/issues/395
 

--- a/utils/litemap/examples/language_names_lite_map.rs
+++ b/utils/litemap/examples/language_names_lite_map.rs
@@ -9,8 +9,23 @@
 
 icu_benchmark_macros::static_setup!();
 
-use litemap::LiteMap;
+use icu_locid::subtags::Language;
 use icu_locid_macros::language;
+use litemap::LiteMap;
+
+const DATA: [(Language, &'static str); 11] = [
+    (language!("ar"), "Arabic"),
+    (language!("bn"), "Bangla"),
+    (language!("ccp"), "Chakma"),
+    (language!("en"), "English"),
+    (language!("es"), "Spanish"),
+    (language!("fr"), "French"),
+    (language!("ja"), "Japanese"),
+    (language!("ru"), "Russian"),
+    (language!("sr"), "Serbian"),
+    (language!("th"), "Thai"),
+    (language!("tr"), "Turkish"),
+];
 
 #[no_mangle]
 fn main(_argc: isize, _argv: *const *const u8) -> isize {
@@ -18,20 +33,12 @@ fn main(_argc: isize, _argv: *const *const u8) -> isize {
 
     let mut map = LiteMap::new();
     // https://github.com/rust-lang/rust/issues/62633 was declined :(
-    map.try_append(language!("ar"), "Arabic").ok_or(()).unwrap_err();
-    map.try_append(language!("bn"), "Bangla").ok_or(()).unwrap_err();
-    map.try_append(language!("ccp"), "Chakma").ok_or(()).unwrap_err();
-    map.try_append(language!("en"), "English").ok_or(()).unwrap_err();
-    map.try_append(language!("es"), "Spanish").ok_or(()).unwrap_err();
-    map.try_append(language!("fr"), "French").ok_or(()).unwrap_err();
-    map.try_append(language!("ja"), "Japanese").ok_or(()).unwrap_err();
-    map.try_append(language!("ru"), "Russian").ok_or(()).unwrap_err();
-    map.try_append(language!("sr"), "Serbian").ok_or(()).unwrap_err();
-    map.try_append(language!("th"), "Thai").ok_or(()).unwrap_err();
-    map.try_append(language!("tr"), "Turkish").ok_or(()).unwrap_err();
+    for (lang, name) in DATA.iter() {
+        map.try_append(lang, name).ok_or(()).unwrap_err();
+    }
 
     debug_assert_eq!(11, map.len());
-    debug_assert_eq!(Some(&"Thai"), map.get(&language!("th")));
+    debug_assert_eq!(Some(&&"Thai"), map.get(&language!("th")));
     debug_assert_eq!(None, map.get(&language!("de")));
 
     0

